WooCommerce/Classes/ViewRelated/Orders/Simple Payments/Amount/SimplePaymentsAmount.swift

```diff
@@ -43,6 +43,14 @@ final class SimplePaymentsAmountHostingController: UIHostingController<SimplePay
                 viewModel.presentNotice = nil
             }
             .store(in: &subscriptions)
+
+        // Observes the loading intent it will be true while performing the create order operation. False otherwise.
+        viewModel.$loading
+            .sink { [weak self] loading in
+                // Disables interactive dismissal while performing the create order operation, Enables otherwise.
+                self?.isModalInPresentation = loading
+            }
+            .store(in: &subscriptions)
```

Comment:
This looks good, another way to achieve this, since we already have a presentation controller delegate is to prevent the view dismissal like this.

```swift
func presentationControllerShouldDismiss(_ presentationController: UIPresentationController) -> Bool {
     !rootView.viewModel.loading
}
```

or even better
```swift
func presentationControllerShouldDismiss(_ presentationController: UIPresentationController) -> Bool {
    !rootView.viewModel.disableCancel
}
```

Reply:
Thank you for pointing out. I have change it to presentation controller delegate method.